MapboxVision/src/test/java/com/mapbox/vision/telemetry/TotalBytesCounterTest.kt

```diff
@@ -0,0 +1,291 @@
+package com.mapbox.vision.telemetry
+
+import com.mapbox.vision.utils.system.Time
+import io.mockk.every
+import io.mockk.mockk
+import java.util.concurrent.TimeUnit
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertFalse
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.spekframework.spek2.Spek
+import org.spekframework.spek2.style.gherkin.Feature
+
+object TotalBytesCounterTest : Spek({
+
+    val TEST_SESSION_MAX_BYTES = 10_000L
+    val TEST_SESSION_LENGTH_MS = TimeUnit.MINUTES.toMillis(10)
+    val CURRENT_TIME = TimeUnit.HOURS.toMillis(20)
+    val DEFAULT_SESSION_SIZE_MB = 30
+
+    Feature("TotalBytesCounter") {
+
+        Scenario("Check default session size") {
+            lateinit var totalBytesCounter: TotalBytesCounter
+            val mockedTime = mockk<Time>()
+
+            Given("TotalBytesCounter with default params") {
+                totalBytesCounter = TotalBytesCounter.Impl(time = mockedTime)
+            }
+
+            var actualValue = false
+
+            When("Get fitInLimit for ${DEFAULT_SESSION_SIZE_MB}MB") {
+                actualValue = totalBytesCounter.fitInLimit(DEFAULT_SESSION_SIZE_MB * 1024 * 1024L)
+            }
+
+            Then("It should be <true>") {
+                assertEquals(true, actualValue)
+            }
+
+            When("Get fitInLimit for ${DEFAULT_SESSION_SIZE_MB}MB and 1 byte") {
+                actualValue = totalBytesCounter.fitInLimit(DEFAULT_SESSION_SIZE_MB * 1024 * 1024L + 1)
+            }
+
+            Then("It should be <false>") {
+                assertEquals(false, actualValue)
+            }
+        }
+
+        Scenario("Check default session length") {
+            lateinit var totalBytesCounter: TotalBytesCounter
+            val mockedTime = mockk<Time>()
+            every { mockedTime.millis() }.returns(0)
+
+            Given("TotalBytesCounter with default params") {
+                totalBytesCounter = TotalBytesCounter.Impl(time = mockedTime)
+            }
+
+            var actualValue = 0L
+
+            When("Get millisToNextSession") {
+                actualValue = totalBytesCounter.millisToNextSession()
+            }
+
+            Then("It should be <1 hour>") {
+                assertEquals(TimeUnit.HOURS.toMillis(1), actualValue)
+            }
+        }
+
+        Scenario("Check fitInLimit") {
```

Comment:
is it scenario exactly for 10 min and 10kBytes? May be rename it to more clean name 
```suggestion
        Scenario("Check fitInLimit 10 min and 10kBytes") {
```

Reply:
discuss. leave current description